### PR TITLE
Hotelcrud mutebinuhu

### PR DESCRIPTION
**Pull request by mutebinuhu**
hello Bonstana have look,

### CODE REVIEW

File: app/Http/Controllers/HotelsController.php [hunk 1]
```diff
@@ -3,83 +3,171 @@
...
-        $hotel->price = $request->get('price');
-        $hotel->save();
-
-
-          return response()->json([
-            "message" => "hotel updated successfully"
-        ], 200);
+     
+        try {
+            $hotel = Hotel::whereid($hotelid)->firstOrFail();
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:108]: Codacy found an issue: [Equals sign not aligned with surrounding assignments; expected 9 spaces but found 1 space](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php [hunk 2]
```diff
@@ -3,83 +3,171 @@
...
 {
-    //
+    //get all hotels
     public function getAllHotels()
     {
-         $hotels = Hotel::get()->toJson(JSON_PRETTY_PRINT);
-        return response($hotels, 200);
+        try {
+
+             $hotels = Hotel::All();
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:19]: Codacy found an issue: [Avoid using static access to class '\App\Hotel' in method 'getAllHotels'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php [hunk 3]
```diff
@@ -3,83 +3,171 @@
...
+    public function deleteHotel(Request $request, $hotelid)
     {
-         $hotel = Hotel::findOrFail($id);
-        $hotel->delete();
-
-        return response()->json([
-            "message" => "hotel deleted"
-        ], 204);
+        try {
+             $hotel = Hotel::findOrFail($hotelid);
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:150]: Codacy found an issue: [Avoid using static access to class '\App\Hotel' in method 'deleteHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php [hunk 4]
```diff
@@ -3,83 +3,171 @@
...
     public function createHotel(Request $request)
     {
-        //validate new hotel
-         $request->validate([
-           'hotel_name'=>'required',
-                'description'=>'required',
-                'price'=>'required|numeric'  
+       try {
+                //validate new hotel
+         $validatedData = Validator::make($request->all(), [
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:62]: Codacy found an issue: [Avoid using static access to class '\Validator' in method 'createHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php [hunk 5]
```diff
@@ -3,83 +3,171 @@
...
-        ], 404);
+
+      try {
+          $hotel = Hotel::whereid($hotelid)->firstOrFail();
+          return response()->json([
+            'success'=>true,
+            'data'=>$hotel
+          ]);
+      } catch (Exception $e) {
+          Log::error('An error occurred while loading hotel'. $e->getMessage());
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:46]: Codacy found an issue: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'getHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php [hunk 6]
```diff
@@ -3,83 +3,171 @@
...
             'price'=>$request->price
             );
-         if (Hotel::create($formdata)) {
-              return response()->json([
-            "message" => "Hotel created"
-        ], 200);
-         }
-         
 
+        Hotel::create($formdata);
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:83]: Codacy found an issue: [Avoid using static access to class '\App\Hotel' in method 'createHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php [hunk 7]
```diff
@@ -3,83 +3,171 @@
...
+            return response()->json([
+                'success'=>true,
+                'data'=>[
+                    'message'=>'hotel updated successfully',
+                    'hotel'=>$hotel
+                ]
+            ]);
+            
+        } catch (\Exception $e) {
+            Log::error('An error occured while updating hotel'. $e->getMessage());
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:136]: Codacy found an issue: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'updateHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php [hunk 8]
```diff
@@ -3,83 +3,171 @@
...
+        Hotel::create($formdata);
+        return response()->json([
+            'success'=>true,
+            'data'=>[
+                'message'=>'hotel created successfully',
+                'hotel'=>$formdata,
+            ]
+        ]);
+       } catch (\Exception $e) {
+           Log::error('An error occurred while creating hotel'.$e->getMessage());
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:92]: Codacy found an issue: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'createHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php [hunk 9]
```diff
@@ -3,83 +3,171 @@
...
-        return response($hotels, 200);
+        try {
+
+             $hotels = Hotel::All();
+             return response()->json([
+                'success'=>true,
+                'data'=>$hotels
+             ]);
+        } catch (\Exception $e) {
+            Log::error('An error occurred while loading hotels' .$e->getMessage());
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:25]: Codacy found an issue: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'getAllHotels'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php [hunk 10]
```diff
@@ -3,83 +3,171 @@
...
+                'data'=>[
+                    'errors'=>'Updating hotel failed please try again'
+                ]
+            ], 500);
+        }
 
     }
-    public function deleteHotel(Request $request, $id)
+    //delete hotel
+    public function deleteHotel(Request $request, $hotelid)
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:147]: Codacy found an issue: [Avoid unused parameters such as '$request'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php [hunk 11]
```diff
@@ -3,83 +3,171 @@
...
+             $hotel->delete();
+             return response()->json([
+                'success'=>true,
+                'data'=>[
+                    'message'=>'hotel successfully deleted',
+                    'hotel'=>$hotel
+                ]
+            ]);
+        } catch (\Exception $e) {
+            Log::error('An error occured while deleting hotel' .$e->getMessage());
```
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:160]: Codacy found an issue: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'deleteHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 6 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:83]: Codacy found an issue: [Avoid using static access to class '\App\Hotel' in method 'createHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 3 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:150]: Codacy found an issue: [Avoid using static access to class '\App\Hotel' in method 'deleteHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 10 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:147]: Codacy found an issue: [Avoid unused parameters such as '$request'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 1 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:108]: Codacy found an issue: [Equals sign not aligned with surrounding assignments; expected 9 spaces but found 1 space](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 7 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:136]: Codacy found an issue: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'updateHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 5 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:46]: Codacy found an issue: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'getHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 2 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:19]: Codacy found an issue: [Avoid using static access to class '\App\Hotel' in method 'getAllHotels'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 8 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:92]: Codacy found an issue: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'createHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 9 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:25]: Codacy found an issue: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'getAllHotels'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 11 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:160]: Codacy found an issue: [Avoid using static access to class '\Illuminate\Support\Facades\Log' in method 'deleteHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

File: app/Http/Controllers/HotelsController.php (hunk 4 above)
Comment (SaidiAli) [inline app/Http/Controllers/HotelsController.php:62]: Codacy found an issue: [Avoid using static access to class '\Validator' in method 'createHotel'.](https://app.codacy.com/gh/BuildForSDGCohort2/Team-225-Backend/pullRequest?prid=6097067)

Comment (highman95) [reply app/Http/Controllers/HotelsController.php:62]: You can use this logic:

$validatedData = $this->validate($request->all(),..........

**Review by SaidiAli — APPROVED** [review]

**Pull request merged**